tests/std/include/test_atomic_wait.hpp

```diff
@@ -0,0 +1,210 @@
+        add_seq('3');
+        a.store(old_value);
+        a.notify_one();
+        std::this_thread::sleep_for(waiting_duration);
+        add_seq('4');
+        a.store(new_value);
+        a.notify_one();
+        // timing assumption that the main thread evaluates the `wait(old_value)` before this timeout expires
+        std::this_thread::sleep_for(waiting_duration);
+        add_seq('6');
+    });
+
+    a.wait(old_value);
+    const auto loaded = a.load();
+    assert(memcmp(&loaded, &new_value, sizeof(UnderlyingType)) == 0);
+
+    add_seq('5');
+
+    thd.join();
+
+    add_seq('\0');
+    assert(strcmp(seq, "123456") == 0);
+}
+
+template <class UnderlyingType>
+void test_notify_all_notifies_all(const UnderlyingType old_value, const UnderlyingType new_value,
+    const std::chrono::steady_clock::duration waiting_duration) {
+    std::atomic<UnderlyingType> c{old_value};
+    const auto waitFn = [&c, old_value] { c.wait(old_value); };
+
+    std::thread w1{waitFn};
+    std::thread w2{waitFn};
+    std::thread w3{waitFn};
+
+    std::this_thread::sleep_for(waiting_duration);
+    c.store(new_value);
+    c.notify_all(); // if this doesn't really notify all, the following joins will deadlock
+
+    w1.join();
+    w2.join();
+    w3.join();
+}
+
+template <class UnderlyingType>
+void test_pad_bits(const std::chrono::steady_clock::duration waiting_duration) {
+    UnderlyingType old_value;
+    std::memset(&old_value, 0x66, sizeof(UnderlyingType));
+    old_value.set(1);
+
+    UnderlyingType same_old_value;
+    std::memset(&same_old_value, 0x99, sizeof(UnderlyingType));
+    same_old_value.set(1);
+
+    std::atomic<UnderlyingType> c(old_value);
+
+    bool trigger      = false;
+    const auto waitFn = [&c, same_old_value, &trigger] {
+        c.wait(same_old_value);
+        trigger = true;
+    };
+
+    std::thread w1{waitFn};
+
+    std::this_thread::sleep_for(waiting_duration);
+    assert(!trigger);
+
+    c.store(old_value);
+    c.notify_one();
+
+    std::this_thread::sleep_for(waiting_duration);
+    assert(!trigger);
+
+    UnderlyingType new_value;
+    std::memset(&new_value, 0x99, sizeof(UnderlyingType));
+    new_value.set(2);
+    c.store(new_value);
+    c.notify_one();
+
+    std::this_thread::sleep_for(waiting_duration);
+    assert(trigger);
+
+    w1.join();
+}
+
+struct two_shorts {
+    short a;
+    short b;
+
+    friend bool operator==(two_shorts, two_shorts) = delete;
+    friend bool operator!=(two_shorts, two_shorts) = delete;
+};
+
+struct three_chars {
+    char a;
+    char b;
+    char c;
+
+    friend bool operator==(three_chars, three_chars) = delete;
+    friend bool operator!=(three_chars, three_chars) = delete;
+};
+
+struct big_char_like {
+    char value;
+    char unused[16];
+
+    explicit big_char_like(char value_) : value(value_), unused{} {}
+
+    big_char_like(const big_char_like&) = default;
+    big_char_like& operator=(const big_char_like&) = default;

```

Comment:
I believe this is unnecessary - providing a constructor doesn't suppress generation of the copy ctor and copy assign. (Whereas providing any ctor suppresses the generation of the default ctor.)